tests/performance-test.rkt

```diff
@@ -0,0 +1,19 @@
+#lang racket/base
+
+(module+ test
+  (require racket/sandbox
+           (prefix-in r1cs: "../picus/r1cs/r1cs-grammar.rkt")
+           (prefix-in cvc5-parser: "../picus/r1cs/r1cs-cvc5-parser.rkt")
+           (prefix-in z3-parser: "../picus/r1cs/r1cs-z3-parser.rkt"))
+
+  ;; read this within 20s or error
+  (define r0
+    (with-limits 20 #f
+      (r1cs:read-r1cs "../benchmarks/r1cs/TreeHasher.r1cs")))
+
+  ;; generate constraints within 20s or error
+  (with-limits 20 #f
+    (call-with-values (λ () (cvc5-parser:parse-r1cs r0 '())) void))
```

Comment:
Why do we have a cvc5 and z3 parser for r1cs?

Reply:
It is actually somewhat misnamed. These are really constraint generators. It's like, we parse R1CS file to generate constraints in Z3/CVC5.

Reply:
As for why there are two of them, my understanding is that we can utilize theory of finite field in CVC5, which is not possible in Z3.

A lot of the code can be shared though. It would be nice to refactor them so that they share common code at some point.

Reply:
I see ok. We should have a parser that parses r1cs into an intermediate representation and then the cvc5 and z3 specific constraint generators can the corresponding constraints.